### ISSUE TITLE
Getting Failed to process message preTest from worker, _0x23bcb9[_0x1d9741(...)] is not a function​​

### ISSUE THREAD

**Issue by djadczak-cksource** [wallabyjs/public#3437]
### Issue description or question

After the recent update to v2 and no changes in either tsconfig, ava.config or wallaby config I started getting:
> ​​[Error] Failed to process message preTest from worker, _0x23bcb9[_0x1d9741(...)] is not a function​​

The test pass when run with ava runner.

### Wallaby diagnostics report

```
{
  editorVersion: '1.95.3',
  pluginVersion: '1.0.399',
  editorType: 'VSCode',
  osVersion: 'darwin 23.4.0',
  nodeVersion: 'v20.11.1',
  coreVersion: '1.0.1661',
  pnp: '<not set>',
  pnpEsm: '<not set>',
  checksum: 'MTBlMzRjZGYyZTFhZTVlOTdjYWZiYTgwNmFlNDEwMWIsMTc2NTQxMTIwMDAwMCww',
  config: {
    files: [
      { pattern: 'src/**/*', ignore: false, trigger: true, load: true, instrument: true, order: 1 },
      { pattern: 'infrastructure/**/*', ignore: false, trigger: true, load: true, instrument: true, order: 2 },
      { pattern: 'assets/on-premises/onpremise_collaboration_public_key.pem', ignore: false, trigger: true, load: true, instrument: true, order: 3 },
      { pattern: 'migrations/**/*.{ts,js,sql}', ignore: false, trigger: true, load: true, instrument: true, order: 4 },
      { pattern: 'tests/migrations/**/*.{ts,js,sql}', ignore: true, trigger: true, load: true, instrument: true, file: true },
      { pattern: 'tests/**/*.{ts,js}', ignore: false, trigger: true, load: true, instrument: true, order: 5 },
      { pattern: 'tests/**/_*.{ts,js}', ignore: false, trigger: true, load: true, instrument: true, order: 6 },
      { pattern: 'tests/unit/_util/**/*.{ts,js}', ignore: false, trigger: true, load: true, instrument: true, order: 7 },
      { pattern: 'tests/unit/**/*.{ts,js}', ignore: true, trigger: true, load: true, instrument: true, file: true },
      { pattern: 'tests/integration/**/*.{ts,js}', ignore: true, trigger: true, load: true, instrument: true, file: true },
      { pattern: 'assets/**/*.{tmpl.html,pem}', ignore: false, trigger: true, load: true, instrument: true, order: 8 },
      { pattern: 'assets/**', ignore: false, trigger: true, load: true, instrument: true, order: 9 }
    ],
    tests: [
      { pattern: 'tests/unit/**/*.{ts,js}', ignore: false, trigger: true, load: true, test: true, order: 10 },
      { pattern: 'tests/unit/repositories/*.{ts,js}', ignore: true, trigger: true, load: true, test: true, file: false }
    ],
    env: {
      type: 'node',
      params: {},
      runner: '<homeDir>/.nvm/versions/node/v20.11.1/bin/node',
      viewportSize: { width: 800, height: 600 },
      options: { width: 800, height: 600 },
      bundle: true
    },
    testFramework: { version: 'ava@1.0.0', configurator: 'ava@1.0.0', reporter: 'ava@1.0.0', starter: 'ava@1.0.0', jsPreprocessor: [Function: jsPreprocessor] },
    preprocessors: { '**/*.jsts': [Function: **/*.jsts] },
    compilers: { '**/*.?(lit)coffee?(.md)': [Function (anonymous)] },
    diagnostics: {},
    captureConsoleLog: true,
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    logLimits: { inline: { depth: 5, elements: 5000 }, values: { default: { stringLength: 8192 }, autoExpand: { elements: 5000, stringLength: 8192, depth: 10 } } },
    maxConsoleMessagesPerTest: 100,
    maxTraceSteps: 999999,
    maxTraceStepsForWatchExpressionPrefetch: 0,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore|c8 ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    mapConsoleMessagesStackTrace: false,
    extensions: {},
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    runAllTestsWhenNoAffectedTests: true,
    symlinkNodeModules: undefined,
    configCode: '/*\n' +
      ' Copyright (c), CKSource Holding sp. z o.o. All rights reserved.\n' +
      ' */\n' +
      "const compilerOptions = require( './tsconfig.json' ).compilerOptions;\n" +
      '\n' +
      'module.exports = wallaby =>\n' +
      '\t( {\n' +
      '\t\tsetup() {\n' +
      '\t\t\tif ( global._tsconfigPathsRegistered ) {\n' +
      '\t\t\t\treturn;\n' +
      '\t\t\t}\n' +
      '\n' +
      "\t\t\tconst tsConfigPaths = require( 'tsconfig-paths' );\n" +
      '\n' +
      '\t\t\ttsConfigPaths.register( {\n' +
      "\t\t\t\tbaseUrl: '.',\n" +
      '\t\t\t\tpaths: {\n' +
      "\t\t\t\t\t'@src': [ 'src' ],\n" +
      "\t\t\t\t\t'@src/*': [ 'src/*' ]\n" +
      '\t\t\t\t}\n' +
      '\t\t\t} );\n' +
      '\n' +
      '\t\t\tglobal._tsconfigPathsRegistered = true;\n' +
      '\t\t},\n' +
      '\t\tfiles: [\n' +
      "\t\t\t'src/**/*',\n" +
      "\t\t\t'infrastructure/**/*',\n" +
      "\t\t\t'assets/on-premises/onpremise_collaboration_public_key.pem',\n" +
      "\t\t\t'migrations/**/*.{ts,js,sql}',\n" +
      '\t\t\t// uncomment this line if tests related to docs fail\n' +
      "\t\t\t// 'assets/**/*',\n" +
      '\t\t\t// remove tests for migration\n' +
      "\t\t\t'!tests/migrations/**/*.{ts,js,sql}',\n" +
      '\t\t\t// assets from tests\n' +
      "\t\t\t'tests/**/*.{ts,js}',\n" +
      '\t\t\t// utils, helpers from tests\n' +
      "\t\t\t'tests/**/_*.{ts,js}',\n" +
      "\t\t\t'tests/unit/_util/**/*.{ts,js}',\n" +
      '\t\t\t// actual test files\n' +
      "\t\t\t'!tests/unit/**/*.{ts,js}',\n" +
      "\t\t\t'!tests/integration/**/*.{ts,js}',\n" +
      "\t\t\t'assets/**/*.{tmpl.html,pem}',\n" +
      "\t\t\t'./assets/**'\n" +
      '\t\t],\n' +
      '\t\ttests: [\n' +
      '\t\t\t// you may want to adjust this section; e.g. include repository tests\n' +
      "\t\t\t'tests/unit/**/*.{ts,js}',\n" +
      "\t\t\t'!tests/unit/repositories/*.{ts,js}'\n" +
      '\t\t],\n' +
      "\t\tenv: { type: 'node' },\n" +
      "\t\ttestFramework: 'ava',\n" +
      "\t\tpreprocessors: { '**/*.jsts': file => file.changeExt( 'js' ).content },\n" +
      "\t\tcompilers: { '**/*.+(t|j)s?(x)': wallaby.compilers.typeScript( compilerOptions ) }\n" +
      '\t} )\n' +
      ';\n'
  },
  packageJSON: {
    dependencies: {
      '@aws-sdk/client-sqs': '3.651.1',
      '@cksource-cs/cks-customer-integration-module': '^31.0.0',
      '@cksource-cs/cks-customer-journey-module': '^3.1.1',
      '@cksource-cs/cks-customer-portal-module': '^25.5.0',
      '@cksource-cs/cks-management-panel-cs-proxy-module': '^1.10.0',
      '@cksource-cs/config-module': '^10.0.9',
      '@cksource-cs/diagnostic-module': '^9.1.5',
      '@cksource-cs/errors-module': '^4.3.56',
      '@cksource-cs/events-module': '8.0.2',
      '@cksource-cs/files-module': '^10.1.0',
      '@cksource-cs/http-client-module': '^1.9.4',
      '@cksource-cs/http-module': '^9.1.0',
      '@cksource-cs/logger-module': '^7.16.2',
      '@cksource-cs/redis-module': '^7.3.1',
      '@cksource-cs/req-res-module': '2.0.0',
      '@cksource-cs/servicediscovery-module': '^4.0.12',
      '@cksource-cs/services-module': '^9.0.5',
      '@cksource-cs/sql-module': '^12.8.6',
      '@cksource-cs/sqs-messaging-module': '^7.0.3',
      '@cksource-cs/utils-module': '^9.8.0',
      '@cksource-cs/validation-module': '^5.0.2',
      '@googleapis/oauth2': '1.0.7',
      bcrypt: '5.1.1',
      cookie: '1.0.1',
      'google-auth-library': '9.14.1',
      handlebars: '4.7.8',
      jsonwebtoken: '9.0.2',
      nanoid: '3.3.6'
    },
    devDependencies: {
      '@cksource-cs/eslint-config-cs-module': '^5.0.0',
      '@cksource-cs/grafana-dashboards-module': '^3.6.3',
      '@cksource-cs/metrics-module': '^3.0.0',
      '@cksource-cs/scheduler-module': '1.0.63',
      '@cksource-cs/tests-module': '^9.1.5',
      '@types/bcrypt': '5.0.2',
      '@types/jsonwebtoken': '9.0.5',
      '@types/node': '20.11.1',
      '@types/sinon': '10.0.16',
      ava: '4.3.3',
      eslint: '8.6.0',
      'eslint-import-resolver-node': '0.3.6',
      'npm-check-updates': '10.2.2',
      nyc: '15.1.0',
      'tsconfig-paths': '3.11.0',
      typescript: '5.4.5'
    }
  },
  fs: { numberOfFiles: 1026 },
  debug: [
    '2024-12-08T13:42:55.285Z project waiting for initial run signal\n',
    '2024-12-08T13:42:55.288Z model Initialization Completed: 115ms\n',
    '2024-12-08T13:42:55.397Z project Wallaby Node version: v20.11.1\n',
    '2024-12-08T13:42:55.397Z project Wallaby config: <homeDir>/Projects/cksource/cs/services/cks/cks-customer-portal/wallaby.ava.unit.js\n',
    '2024-12-08T13:42:55.429Z fs File system starting\n',
    '2024-12-08T13:42:56.520Z fs File system scan has finished by timeout\n',
    '2024-12-08T13:42:56.539Z project File cache: <homeDir>/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.399/projects/9a6aab3a1de5d65a\n',
    '2024-12-08T13:42:56.580Z ui.service.legacy Listening port 51235\n',
    '2024-12-08T13:42:56.580Z project Config file change detected, invalidating local cache\n',
    '2024-12-08T13:42:56.659Z workers Parallelism for initial run: 8, for regular run: 4\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #0\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #1\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #2\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #3\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #4\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #5\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #6\n',
    '2024-12-08T13:42:56.659Z workers Starting run worker instance #7\n',
    '2024-12-08T13:42:56.659Z workers Web server is listening at 52026\n',
    '2024-12-08T13:42:56.659Z project File cache requires some updates, waiting required files from IDE\n',
    '2024-12-08T13:42:56.695Z fs File changed in editor: src/subscriptions/SelfServiceSubscriptionProductsMapper.ts\n',
    '2024-12-08T13:42:56.697Z fs File changed in editor: src/licenses/commands/generateLicenseKeys/GenerateLicenseKeysCommandHandler.ts\n',
    '2024-12-08T13:42:56.697Z fs File changed in editor: src/repositories/queries/usage/mysql/getSubscriptionUsageQuery.ts\n',
    '2024-12-08T13:42:56.697Z fs File changed in editor: src/organizations/queries/getUserOrganizations/GetUserOrganizationsServiceHandler.ts\n',
    '2024-12-08T13:42:56.697Z fs File changed in editor: src/subscriptions/queries/getSubscriptionDetails/SubscriptionDTO.ts\n',
    '2024-12-08T13:42:56.891Z workers Started run worker instance (delayed) #3\n',
    '2024-12-08T13:42:56.929Z workers Started run worker instance (delayed) #0\n',
    '2024-12-08T13:42:56.941Z workers Started run worker instance (delayed) #2\n',
    '2024-12-08T13:42:56.950Z workers Started run worker instance (delayed) #4\n',
    '2024-12-08T13:42:56.973Z workers Started run worker instance (delayed) #1\n',
    '2024-12-08T13:42:56.991Z workers Started run worker instance (delayed) #6\n',
    '2024-12-08T13:42:56.994Z workers Started run worker instance (delayed) #5\n',
    '2024-12-08T13:42:56.997Z workers Started run worker instance (delayed) #7\n',
    '2024-12-08T13:42:57.298Z project Stopping process pool\n',
    '2024-12-08T13:42:57.302Z project Running postprocessor\n',
    '2024-12-08T13:42:57.321Z postprocessor New TypeScript language service is required\n',
    '2024-12-08T13:43:03.383Z project Postprocessor execution finished\n',
    '2024-12-08T13:43:03.383Z project Test run started; run priority: 3\n',
    '2024-12-08T13:43:03.385Z project Running all tests\n',
    '2024-12-08T13:43:03.397Z workers Starting test run, priority: 3\n',
    '2024-12-08T13:43:03.397Z workers Distributing tests between 8 workers\n',
    '2024-12-08T13:43:03.403Z workers Running tests in parallel\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #0, session #iil49]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #1, session #gtw2d]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #2, session #mfukq]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #3, session #lbmyd]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #4, session #b96iq]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #5, session #i0qeq]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #6, session #nigqz]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Starting sandbox [worker #7, session #scqf0]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #0, session #iil49]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #1, session #gtw2d]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #2, session #mfukq]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #3, session #lbmyd]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #4, session #b96iq]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #5, session #i0qeq]\n',
    '2024-12-08T13:43:03.403Z nodeRunner Preparing sandbox [worker #6, session #nigqz]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Preparing sandbox [worker #7, session #scqf0]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #0, session #iil49]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #1, session #gtw2d]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #2, session #mfukq]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #3, session #lbmyd]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #4, session #b96iq]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #5, session #i0qeq]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #6, session #nigqz]\n',
    '2024-12-08T13:43:03.404Z nodeRunner Prepared sandbox [worker #7, session #scqf0]\n',
    '2024-12-08T13:43:03.404Z workers [worker #0, session #iil49] Running tests in sandbox\n',
    '2024-12-08T13:43:03.405Z workers [worker #1, session #gtw2d] Running tests in sandbox\n',
    '2024-12-08T13:43:03.405Z workers [worker #2, session #mfukq] Running tests in sandbox\n',
    '2024-12-08T13:43:03.405Z workers [worker #3, session #lbmyd] Running tests in sandbox\n',
    '2024-12-08T13:43:03.406Z workers [worker #4, session #b96iq] Running tests in sandbox\n',
    '2024-12-08T13:43:03.406Z workers [worker #5, session #i0qeq] Running tests in sandbox\n',
    '2024-12-08T13:43:03.406Z workers [worker #6, session #nigqz] Running tests in sandbox\n',
    '2024-12-08T13:43:03.407Z workers [worker #7, session #scqf0] Running tests in sandbox\n',
    '2024-12-08T13:43:03.508Z workers [gtw2d] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.509Z workers [i0qeq] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.509Z workers [lbmyd] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.511Z workers [nigqz] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.513Z workers [iil49] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.513Z workers [scqf0] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.514Z workers [mfukq] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.516Z workers [b96iq] Loaded unknown number of test(s)\n',
    '2024-12-08T13:43:03.987Z workers [gtw2d] Sandbox is responsive, closing it\n',
    '2024-12-08T13:43:03.990Z project Test run finished\n',
    '2024-12-08T13:43:04.052Z workers [lbmyd] Sandbox is not responsive, recycling worker instance\n',
    '2024-12-08T13:43:04.052Z workers Recyling run worker instance #3\n',
    '2024-12-08T13:43:04.055Z workers [nigqz] Sandbox is responsive, closing it\n',
    '2024-12-08T13:43:04.055Z workers [b96iq] Sandbox is responsive, closing it\n',
    '2024-12-08T13:43:04.057Z workers [mfukq] Sandbox is not responsive, recycling worker instance\n',
    '2024-12-08T13:43:04.057Z workers Recyling run worker instance #2\n',
    '2024-12-08T13:43:04.058Z workers [scqf0] Sandbox is responsive, closing it\n',
    '2024-12-08T13:43:04.178Z workers [iil49] Sandbox is responsive, closing it\n',
    '2024-12-08T13:43:04.215Z workers [i0qeq] Sandbox is responsive, closing it\n',
    '2024-12-08T13:43:11.723Z ui.service.legacy UI client connected\n',
    '2024-12-08T13:43:11.723Z ui.service.legacy Outgoing message ui:handshake\n',
    '2024-12-08T13:43:11.731Z ui.service.legacy Incoming message ui:start\n',
    '2024-12-08T13:43:11.732Z ui.service.legacy Outgoing message ui:summary\n',
    '2024-12-08T13:43:11.733Z ui.service.legacy Outgoing message ui:files\n',
    '2024-12-08T13:43:11.734Z ui.service.legacy Incoming message ui:tests:resultsRequested\n',
    '2024-12-08T13:43:11.734Z ui.service.legacy Outgoing message ui:tests:allResultsUpdated\n'
  ]
}
```

**Comment by smcenlly** [comment]
Thanks for reporting the problem. We inadvertently introduced this problem in our Ava integration as a part of Wallaby v2. This is fixed in Wallaby core `v1.0.1662`.

Wallaby should automatically update for you, but if it does not, you may run the VS Code command, `Wallaby.js: Force Core Update`.

**Comment by djadczak-cksource** [comment]
Beautiful , I confirm it works.

You guys rock, A Star support, A Star product, A start testing with wallaby :) 

Thanks!

**Comment by djadczak-cksource** [comment]
Not sure if connected , but I am having issues running the Debugger as well.
No extra logs from what I've initially looked. Will try to dump analytics after work, but go with console.logs and `.only` for now.

**Comment by smcenlly** [comment]
Thanks for letting us know that the Debugger was not working for you. It is fixed in the latest version of Wallaby core `v1.0.1664`.
